### PR TITLE
Remove --rbac flag

### PR DESCRIPTION
**Pull request by suaas21**
xref: #705

Tasks:
- [x] remove --rbac flag
- [x] remove --rbac flag from helm chart 
- [x] run `./hack/gendocs/make.sh`

### CODE REVIEW

File: chart/stash/values.yaml [hunk 1]
```diff
@@ -53,11 +53,6 @@ tolerations: {}
 ##
 affinity: {}
 
-## Install Default RBAC roles and bindings
```
Comment (tamalsaha) [inline chart/stash/values.yaml]: Update table
https://github.com/stashed/stash/blob/master/chart/stash/README.md#configuration

File: pkg/cmds/server/options.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ func NewExtraOptions() *ExtraOptions {
 		QPS:            100,
 		Burst:          100,
 		ResyncPeriod:   10 * time.Minute,
+		EnableRBAC:     true,
```
Comment (tamalsaha) [inline pkg/cmds/server/options.go]: Remove EnableRBAC variable. Just assume it is true.

File: chart/stash/values.yaml (hunk 1 above)
Comment (suaas21) [reply chart/stash/values.yaml]: resolved

File: pkg/cmds/server/options.go (hunk 2 above)
Comment (suaas21) [reply pkg/cmds/server/options.go]: removed EnableRBAC variable

File: test/e2e/framework/functions.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func (f *Invocation) UpdateStatusFunction() v1beta1.Function {
 				fmt.Sprintf("--backup-session=${%s:=}", apis.BackupSession),
 				fmt.Sprintf("--restore-session=${%s:=}", apis.RestoreSession),
 				fmt.Sprintf("--output-dir=${%s:=}", outputDir),
-				fmt.Sprintf("--enable-status-subresource=${%s:=%s}", apis.StatusSubresourceEnabled, apis.StatusSubresourceEnabled),
+				fmt.Sprintf("--enable-status-subresource=${%s:=false}", apis.StatusSubresourceEnabled),
```
Comment (tamalsaha) [inline test/e2e/framework/functions.go]: Should be default true. Why change this?

File: test/e2e/framework/statefulset.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ func (fi *Invocation) StatefulSetForV1beta1API() apps.StatefulSet {
 						StorageClassName: types.StringP(fi.StorageClass),
 						Resources: core.ResourceRequirements{
 							Requests: core.ResourceList{
-								core.ResourceName(core.ResourceStorage): resource.MustParse("1Gi"),
+								core.ResourceName(core.ResourceStorage): resource.MustParse("10Gi"),
```
Comment (tamalsaha) [inline test/e2e/framework/statefulset.go]: This feels too high for minikube.

File: test/e2e/framework/functions.go (hunk 3 above)
Comment (suaas21) [reply test/e2e/framework/functions.go]: fixed

File: test/e2e/framework/statefulset.go (hunk 4 above)
Comment (suaas21) [reply test/e2e/framework/statefulset.go]: fixed

**Pull request merged**